stable_baselines/td3/td3.py

```diff
@@ -294,121 +294,121 @@ def learn(self, total_timesteps, callback=None, seed=None,
             start_time = time.time()
             episode_rewards = [0.0]
             episode_successes = []
+            obs = self.env.reset()
             if self.action_noise is not None:
                 self.action_noise.reset()
-            obs = self.env.reset()
             self.episode_reward = np.zeros((1,))
             ep_info_buf = deque(maxlen=100)
             n_updates = 0
             infos_values = []
 
-            for step in range(total_timesteps):
+            for step in range(0, total_timesteps, self.n_envs):
                 if callback is not None:
                     # Only stop training if return value is False, not when it is None. This is for backwards
                     # compatibility with callbacks that have no return statement.
                     if callback(locals(), globals()) is False:
                         break
 
-                # Before training starts, randomly sample actions
-                # from a uniform distribution for better exploration.
-                # Afterwards, use the learned policy
-                # if random_exploration is set to 0 (normal setting)
+                prev_obs = obs
                 if (self.num_timesteps < self.learning_starts
-                        or np.random.rand() < self.random_exploration):
+                    or np.random.rand() < self.random_exploration):
                     # No need to rescale when sampling random action
-                    rescaled_action = action = self.env.action_space.sample()
+                    rescaled_action = action = [self.env.action_space.sample() for _ in range(self.n_envs)]
                 else:
-                    action = self.policy_tf.step(obs[None]).flatten()
+                    action = self.policy_tf.step(prev_obs).flatten()
+                    action = [np.array([a]) for a in action]
```

Comment:
why not removing the flatten instead?

Reply:
I think I've removed this line from the code (the red highlight indicates that).

Reply:
Sounds good